### ISSUE TITLE
PR for 4.3.3-beta -- aiming for GA

### ISSUE THREAD

**Issue by narategithub** [ovis-hpc/ovis#94]
Since we are currently aiming for GA, the Pull Requests (PRs) that are not bug fixes are postponed to be merged + tested after 4.3.3 has released.

The following PRs are to be included in the test branch + test:

These will be merged if it passed the test.
#70, #83, #84, #86, #92

The following PRs are to be staged + tested later:
#72, #75, #76, #78, #79, #82, #85, #88, #91

Please feel free to voice your opinion if you need some items to be added before GA :)

p.s. Since I do not know how to tag `everybody`, I'll have to tag you all manually:
@valleydlr @oceandlr @baallan @bschwal @aazizo @nichamon @narategithub @tom95858 @morrone

**Comment by baallan** [comment]
From my comments on issue 94,

"I would like to get our house (names of libraries, locations of libraries, number of configure scripts) in order before we leave beta so that anyone creating plugins against a post-beta (GA) v4 release will not have to rename libraries in their linker flags. There shouldn't be .so file name changes (other than possibly adding new .so files) for the life of v4."

This would be a good time for folks aiming to create plugins to tell us what they think about the meaning of GA. Does code in GA evolve its library names from one release to the next, e.g.
from 4.3.3 to 4.3.4?
from 4.3.x to 4.4.0?

**Comment by narategithub** [comment]
The staged branch passed the minimal build (no options given) and 4 sampler => 2xLV1 agg => 1xLV2 agg test. So, I pushed them to OVIS-4.3.3-beta.

@tom95858 The test code is in `ldms-test/agg_test`.

**Comment by morrone** [comment]
@narategithub, I think that #88 can probably land too.

I'm fine with my PRs that are in the "later" list going in later. I'm just hoping that "later" is before the end of 2019, not 6 months from now.

**Comment by morrone** [comment]
@baallan I would recommend that ovis adopt the Semantic Versioning scheme, and the general rules that it suggests for incrementing numbers.

 https://semver.org/

This approach is common,in fact it was in common usage long before "Semantic Versioning" was codified based on those common approaches.  Since this approach is common, rpms (and probably other packaging systesm) have guidelines for rpm versioning that handle mapping Semantic Versioning to rpm versions pretty well:

 https://docs.fedoraproject.org/en-US/packaging-guidelines/Versioning/

But I would definitely like to see the "get our house in order" stuff happen before we start enforcing those rules. I'm one of the people that will have to handle the changing library names in an external package, and I think it is totally worth it.

**Comment by oceandlr** [comment]
@morrone @narategithub @tom95858 @valleydlr 
Please see new comment in #83. I think there is something wrong we will need to fix. @narategithub why were these pulled into master and not integration-and-test branch? Or did you already test?

**Comment by morrone** [comment]
Yes, the bad EXTRA_DIST would not have been obvious in normal testing.

**Comment by ovis-hpc** [comment]
@narategithub @tom95858 @valleydlr @morrone 
@oceandlr merged #96 (fix discussed above including rebased #88 ) into master

**Comment by baallan** [comment]
@morrone As discussed at ldmscon v4.3 violates semantic versioning practices for legal reasons. We are aiming to get more in line with semver, but aren't there yet.

**Comment by oceandlr** [comment]
OVIS-4.3.3-RC1 has been released

**Comment by oceandlr** [comment]
@narategithub @morrone @baallan OVIS-4.3.3 GA has been released. Can we close this issue since this was about requests to go into that version?

**Comment by narategithub** [comment]
sure :)
